docker-versions-maven-plugin/src/main/java/com/github/cafapi/docker_versions/docker/auth/DockerAuthConfig.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2024 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.docker_versions.docker.auth;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Arrays;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+final class DockerAuthConfig
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(DockerAuthConfig.class);
+
+    private DockerAuthConfig()
+    {
+    }
+
+    public static DockerRegistryAuthConfig getRegistryAuthConfig(final String registry) throws DockerRegistryAuthException
+    {
+        final ObjectNode dockerConfig;
+        try {
+            dockerConfig = readDockerConfig();
+        } catch (final IOException e) {
+            LOGGER.warn("Error reading docker config", e);
+            return null;
+        }
+        if (dockerConfig == null) {
+            return null;
+        }
+        final String registryToLookup;
+        if (registry != null
+            && Arrays.stream(Constants.DEFAULT_REGISTRIES).noneMatch(r -> r.equalsIgnoreCase(registry))) {
+            registryToLookup = registry;
+        } else {
+            registryToLookup = Constants.DEFAULT_DOCKER_REGISTRY;
+        }
+
+        DockerRegistryAuthConfig authConfig;
+        if (dockerConfig.has("credHelpers")) {
+            final JsonNode credHelpers = dockerConfig.get("credHelpers");
+            if (credHelpers.has(registryToLookup)) {
+                authConfig = extractAuthConfigFromCredentialsHelper(
+                    registryToLookup,
+                    credHelpers.get(registryToLookup).asText());
+                if (authConfig != null) {
```

Comment:
Don't fallback from credential helpers when the credential helper has been registered for the auth key.